src/test/java/it/gov/pagopa/admissibility/event/processor/AdmissibilityProcessorConfigRankingTest.java

```diff
@@ -0,0 +1,372 @@
+package it.gov.pagopa.admissibility.event.processor;
+
+import com.azure.spring.messaging.AzureHeaders;
+import com.azure.spring.messaging.checkpoint.Checkpointer;
+import com.azure.spring.messaging.servicebus.support.ServiceBusMessageHeaders;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import it.gov.pagopa.admissibility.BaseIntegrationTest;
+import it.gov.pagopa.admissibility.dto.onboarding.OnboardingDTO;
+import it.gov.pagopa.admissibility.dto.onboarding.RankingRequestDTO;
+import it.gov.pagopa.admissibility.dto.rule.Initiative2BuildDTO;
+import it.gov.pagopa.admissibility.event.consumer.BeneficiaryRuleBuilderConsumerConfigIntegrationTest;
+import it.gov.pagopa.admissibility.repository.InitiativeCountersRepository;
+import it.gov.pagopa.admissibility.service.onboarding.*;
+import it.gov.pagopa.admissibility.test.fakers.Initiative2BuildDTOFaker;
+import it.gov.pagopa.admissibility.test.fakers.OnboardingDTOFaker;
+import it.gov.pagopa.admissibility.utils.TestUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.TopicPartition;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.data.util.Pair;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.test.context.TestPropertySource;
+import reactor.core.Scannable;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.IntStream;
+
+@Slf4j
+@TestPropertySource(properties = {
+        "app.beneficiary-rule.build-delay-duration=PT1S",
+        "logging.level.it.gov.pagopa.admissibility.service.build=WARN",
+        "logging.level.it.gov.pagopa.admissibility.service.onboarding=WARN",
+        "logging.level.it.gov.pagopa.admissibility.service.onboarding.AdmissibilityEvaluatorMediatorServiceImpl=WARN",
+        "logging.level.it.gov.pagopa.admissibility.service.BaseKafkaConsumer=WARN",
+})
+class AdmissibilityProcessorConfigRankingTest extends BaseIntegrationTest {
```

Comment:
could we create a Base test with common logic between this test and AdmissibilityProcessorConfigTest?

The @TestConfiguration class probably should be defined inside the subClass, but let's try letting it inside the superClass, in case of errors, in the subclass let's create a @TestConfiguration that just extend the static class defined in the superclass